### PR TITLE
Setup Learning to Rank

### PR DESCRIPTION
**Pull request by jdddog**
This PR adds Learning to Rank to the DMP works search pipeline:
* Updated `rank-metrics` command to compute ranking metrics directly from OpenSearch rather than a file, to save the results in a CSV and compute metrics for each DMP (so we can see which DMPs are working well and which are not).
* Added `create-featureset` command to create featureset in OpenSearch.
* Added `generate-training-dataset` command to generate a training dataset from the ground truth data, a candidate search and the featureset. The RankLib file that is saved is then used to train a RankLib model.
* Added `upload-ranklib-model` command to upload the trained RankLib model to OpenSearch, including feature normalization settings.
* Updated the DMP works search command so that a model name can be supplied, and if it is, the results will be rescored.

### CODE REVIEW

**Review by briri — COMMENTED** [review]
This looks good @jdddog.

I just have a request to add a `.env.example` file to the repo, and as I mentioned in our meeting, just a screenshot maybe of the ranking metrics (redact/obscure any DOIs) and a writeup on how to evaluate or assess when the model has improved since the last run.

Can you also, when you have a moment, update [this ticket](https://github.com/CDLUC3/dmptool-infrastructure/issues/17)? This is work I am planning to do to facilitate search for DMPs within the UI and API. Those require faceting and search, so might actually need to be a different OS index than the one you're using for DMPs. Anyway, if you could add the list the name of the index and the fields it needs on that ticket, I would appreciate it. Once that work is done, it will negate the need to generate the DMP S3 file.

**Comment by jdddog** [comment]
> This looks good @jdddog.

Thanks @briri,

> 
> I just have a request to add a `.env.example` file to the repo, and as I mentioned in our meeting, just a screenshot maybe of the ranking metrics (redact/obscure any DOIs) and a writeup on how to evaluate or assess when the model has improved since the last run.

Yeah sure, I'm also reviewing the docs at the same time to make sure that I haven't forgotten anything. 

> Can you also, when you have a moment, update [this ticket](https://github.com/CDLUC3/dmptool-infrastructure/issues/17)? This is work I am planning to do to facilitate search for DMPs within the UI and API. Those require faceting and search, so might actually need to be a different OS index than the one you're using for DMPs. Anyway, if you could add the list the name of the index and the fields it needs on that ticket, I would appreciate it. Once that work is done, it will negate the need to generate the DMP S3 file.

Yeah that makes sense, I've left a comment on the other ticket. Y I'll restructure the documentation a bit
